util/mathutil/exponential_average.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mathutil
+
+import "sync"
+
+func factor(n int) float64 {
+	return 2.0 / float64(n+1)
+}
+
+// ExponentialAverage is an exponential average measurement implementation.
+type ExponentialAverage struct {
+	value        float64
+	sum          float64
+	window       int
+	warmupWindow int
+	count        int
+
+	mu sync.RWMutex
+}
+
+// NewExponentialAverage will create a new ExponentialAverage
+func NewExponentialAverage(
+	window int,
+	warmupWindow int,
+) *ExponentialAverage {
+	return &ExponentialAverage{
+		window:       window,
```

Comment:
only some naming concerns, s.window does not have a clear meaning? decay, decreasing factor or something has a clear meaning, but we should also add a check about 0<f<=1

And ExponentialAverage is not familiar as ExponentialMovingAverage, EMAverage or ExpMovingAvg, etc

Reply:
Sure, it is called ```factor```，more appropriately.